rules/Privatization/Guard/ParentPropertyLookupGuard.php

```diff
@@ -51,8 +45,9 @@ public function isLegal(Property $property, Class_ $class): bool
             return false;
         }
 
-        if (! $class->extends instanceof Name) {
-            return true;
+        $nativeReflection = $classReflection->getNativeReflection();
+        if (! $nativeReflection->getParentClass()) {
```

Comment:
extends class may be not autoloaded, so need native reflection to ensure it has parent class.

Reply:
It seems the NativeReflection is BetterReflection class, so it got test error on not autoload part:

```diff
 final class KeepParentNotAutoloaded extends NotAutoloadedParent
 {
-    protected $value = 100;
+    private $value = 100;
 }
```

Reply:
Fixed by pull from `BetterReflectionClass` object https://github.com/rectorphp/rector-src/pull/3993/commits/36f291f0d490cad21d6d4e32b3e5aa5f411ac23c